extensions/amp-ad-network-adsense-impl/0.1/test/test-amp-ad-network-adsense-impl.js

```diff
@@ -99,4 +100,28 @@ describe('amp-ad-network-adsense-impl', () => {
       });
     });
   });
+
+  describe('#getAdUrl', () => {
+    it('returns the right URL', () => {
+      adsenseImpl.onLayoutMeasure();
+      return adsenseImpl.getAdUrl().then(url => {
+        expect(url).to.match(new RegExp(
+          'https://googleads\\.g\\.doubleclick\\.net/pagead/ads' +
+          '\\?client=adsense&format=0x0&w=0&h=0&adtest=false' +
+          '&adk=4075575999&bc=1&vis=1&wgl=1' +
+          '&is_amp=3&amp_v=%24internalRuntimeVersion%24' +
+          // Depending on how the test is run, it can get different results.
+          '&d_imp=1&dt=[0-9]+&ifi=[0-9]+&adf=1597394791' +
+          '&c=[0-9]+&output=html&nhd=1&biw=[0-9]+&bih=[0-9]+' +
+          '&adx=-10000&ady=-10000&u_aw=[0-9]+&u_ah=[0-9]+&u_cd=24' +
+          '&u_w=[0-9]+&u_h=[0-9]+&u_tz=-?[0-9]+&u_his=[0-9]+' +
+          '&brdim=[0-9]+(%2C[0-9]+){9}' +
+          '&isw=[0-9]+&ish=[0-9]+&dtd=[0-9]+' +
+          '&url=https?%3A%2F%2F[a-zA-Z0-9.:%]+' +
+          '&top=https?%3A%2F%2Flocalhost%3A9876%2F%3Fid%3D[0-9]+' +
+          '(&loc=https?%3A%2F%2[a-zA-Z0-9.:%]+)?' +
+          '&ref=https?%3A%2F%2Flocalhost%3A9876%2F%3Fid%3D[0-9]+'));
+      });
```

Comment:
(Edit: You would need to trim off the stuff before the query string first, of course.)